src/MarDyn.cpp

```diff
@@ -132,8 +132,20 @@ int run_unit_tests(const Values &options, const std::vector<std::string> &args)
  */
 int main(int argc, char** argv) {
 #ifdef ENABLE_MPI
-	MPI_Init(&argc, &argv);
-#endif
+#if defined(_OPENMP)
+	const int thread_support_level_requested = MPI_THREAD_FUNNELED; /* So far only the main thread makes MPI calls */
```

Comment:
Out of interest:
How do you know that in ls1 on the main thread makes MPI calls? Because nowhere in the code there is `#pragma omp parallel` and MPI combined?

Reply:
I went through the codebase looking for all places with `#pragma omp parallel`...
Nearly all MPI stuff is encapsulated and outside the loops that use OpenMP. The only place I found where MPI might be affected by OpenMP was in the `NonBlockingMPIMultiStepHandler` if `ADVANCED_OVERLAPPING` is enabled. However, here it is protected with a respective `omp master` ensuring that only the primary thread runs this part.

Also, if there was any problematic place, the code would have had issues with the existing `MPI_Init`. ;)